draft-ietf-quic-recovery.md

```diff
@@ -518,14 +518,17 @@ or when Initial or Handshake keys are discarded. This ensures the PTO is always
 set based on the latest RTT information and for the last sent packet in the
 correct packet number space.
 
+When setting the PTO timer, the ApplicationData packet number space (Section
+4.1.1 of {{QUIC-TLS}}) MUST be ignored until the handshake completes. Not arming
+the PTO for ApplicationData prevents a client from retransmitting a 0-RTT packet
+on a PTO expiration before confirming that the server is able to decrypt 0-RTT
+packets, and prevents a server from sending a 1-RTT packet on a PTO expiration
+before the client obtains the keys to decrypt the 1-RTT packet (Section 4.1.4 of
+{{QUIC-TLS}}) or before the server obtains the keys to process an
+acknowledgement.
+
 When ack-eliciting packets in multiple packet number spaces are in flight,
-the timer MUST be set for the packet number space with the earliest timeout,
-with one exception. The ApplicationData packet number space (Section 4.1.1
-of {{QUIC-TLS}}) MUST be ignored until the handshake completes. Not arming
-the PTO for ApplicationData prevents a client from retransmitting a 0-RTT
-packet on a PTO expiration before confirming that the server is able to
-decrypt 0-RTT packets, and prevents a server from sending a 1-RTT packet on
-a PTO expiration before it has the keys to process an acknowledgement.
+the timer MUST be set for the packet number space with the earliest timeout.
```

Comment:
This now seems to contradict the previous paragraph.

Reply:
Yeah technically we can say that "the timer MUST be set for the (non-ignored) packet number space with the earliest timeout." But does that make a difference?

Reply:
It probably pays to be precise in this case.  You could instead say that you set the Initial or Handshake PTO timer based on which space has the earliest timeout.

Reply:
Thank you for the suggestion. Applied in 3e3421b.